usr.bin/ranlib/ranlib.1
.\" Copyright (c) 1990 Regents of the University of California.
.\" All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\" 3. All advertising materials mentioning features or use of this software
.\"    must display the following acknowledgement:
.\"	This product includes software developed by the University of
.\"	California, Berkeley and its contributors.
.\" 4. Neither the name of the University nor the names of its contributors
.\"    may be used to endorse or promote products derived from this software
.\"    without specific prior written permission.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
.\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
.\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
.\" ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
.\" FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
.\" DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
.\" OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
.\" HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
.\" LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
.\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
.\" SUCH DAMAGE.
.\"
.\"     from: @(#)ranlib.1	6.7 (Berkeley) 5/9/91
.\"	$Id: ranlib.1,v 1.1 1995/10/18 08:45:58 deraadt Exp $
.\"
.Dd May 9, 1991
.Dt RANLIB 1
.Os
.Sh NAME
.Nm ranlib
.Nd table-of-contents for archive libraries
.Sh SYNOPSIS
.Nm ranlib
.Op Fl t
.Ar file ...
.Sh DESCRIPTION
.Nm Ranlib
creates a table of external references for archive libraries,
normally used by the loader,
.Xr ld 1 .
This table is is named ``__.SYMDEF'' and is prepended to the archive.
Files in the archive which are not executable and symbols which are
uninteresting to the loader are ignored.
.Pp
The options are as follows:
.Bl -tag -width indent
.It Fl t
Set the modification time of the __.SYMDEF file.
This time is compared by the loader with the modification time of the
archive to verify that the table is up-to-date with respect to the
archive.
If the modification time has been changed without any change to the
archive (for example, by a
.Xr cp 1 ) ,
the
.Fl t
option can be used to ``touch'' the modification time so that it
appears that the table is up-to-date.
This is also useful after using the
.Fl t
option of
.Xr make 1 .
.El
.Sh ENVIRONMENT
.Bl -tag -width TMPDIR -compact
.It Ev TMPDIR
The pathname of the directory to use when creating temporary files.
.El
.Sh FILES
.Bl -tag -width /tmp/ranlib.XXXXXX -compact
.It Pa /tmp/ranlib.XXXXXX
Temporary file names.
.El
.Sh SEE ALSO
.Xr ar 1 ,
.Xr ld 1 ,
.Xr lorder 1 ,
.Xr nm 1 ,
.Xr ranlib 5
.Sh HISTORY
A
.Nm ranlib
command appeared in
.At v7 .


usr.bin/ranlib/ranlib.c
/*-
 * Copyright (c) 1990 The Regents of the University of California.
 * All rights reserved.
 *
 * This code is derived from software contributed to Berkeley by
 * Hugh Smith at The University of Guelph.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 3. All advertising materials mentioning features or use of this software
 *    must display the following acknowledgement:
 *	This product includes software developed by the University of
 *	California, Berkeley and its contributors.
 * 4. Neither the name of the University nor the names of its contributors
 *    may be used to endorse or promote products derived from this software
 *    without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
 * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 * SUCH DAMAGE.
 */

#ifndef lint
char copyright[] =
"@(#) Copyright (c) 1990 The Regents of the University of California.\n\
 All rights reserved.\n";
#endif /* not lint */

#ifndef lint
/*static char sccsid[] = "from: @(#)ranlib.c	5.6 (Berkeley) 2/26/91";*/
static char rcsid[] = "$Id: ranlib.c,v 1.1 1995/10/18 08:45:58 deraadt Exp $";
#endif /* not lint */

#include <sys/types.h>
#include <dirent.h>
#include <stdio.h>
#include <stdlib.h>
#include <archive.h>

CHDR chdr;
u_int options;				/* UNUSED -- keep open_archive happy */
char *archive;

main(argc, argv)
	int argc;
	char **argv;
{
	int ch, eval, tflag;

	tflag = 0;
	while ((ch = getopt(argc, argv, "t")) != EOF)
		switch(ch) {
		case 't':
			tflag = 1;
			break;
		case '?':
		default:
			usage();
		}
	argc -= optind;
	argv += optind;

	if (!*argv)
		usage();

	for (eval = 0; archive = *argv++;)
		eval |= tflag ? touch() : build();
	exit(eval);
}

usage()
{
	(void)fprintf(stderr, "usage: ranlib [-t] archive ...\n");
	exit(1);
}
